Source/CarthageKit/Errors.swift

```diff
@@ -334,6 +337,19 @@ extension CarthageError: CustomStringConvertible {
 
 		case let .internalError(description):
 			return description
+
+		case let .invalidResolvedCartfile(incompatibilities):
+			var message = "The following incompatibilities were found in Cartfile.resolved:\n"
+			var lines: [String] = []
+			let sortedIncompatibilities = incompatibilities.sorted(by: { $0.dependency.name < $1.dependency.name })
+			sortedIncompatibilities.forEach { incompatibility in
+				let sortedRequirements = incompatibility.incompatibleRequirements.sorted(by: { $0.0.name < $1.0.name })
+				for (dependency, version) in sortedRequirements {
+					lines.append("* \(incompatibility.dependency.name) \(incompatibility.pinnedVersion) is incompatible with \(dependency.name) \(version)")
+				}
+			}
+			message += lines.joined(separator: "\n")
+			return message
```

Comment:
I think this could be written more clearly in a functional style:

```swift
let lines = incompatibilities
  .sorted { $0.dependency.name < $1.dependency.name }
  .flatMap { incompatibility in
    let sortedRequirements = incompatibility
      .incompatibleRequirements
      .sorted { $0.0.name < $1.0.name } }
    return sortedRequirements.map { dependency, version in
      return "* \(incompatibility.dependency.name) \(incompatibility.pinnedVersion) is incompatible with \(dependency.name) \(version)"
    }
  }
  .joined(separator: "\n")
```